addons/portal/models/portal_mixin.py

```diff
@@ -1,22 +1,31 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
-
+import uuid
 from werkzeug.urls import url_encode
-
-from odoo import api, fields, models, _
+from odoo import api, exceptions, fields, models, _
 
 
 class PortalMixin(models.AbstractModel):
     _name = "portal.mixin"
 
-    portal_url = fields.Char(
-        'Portal Access URL', compute='_compute_portal_url',
+    access_url = fields.Char(
+        'Portal Access URL', compute='_compute_access_url',
         help='Customer Portal URL')
+    access_token = fields.Char(
+        'Security Token', copy=False,
+        default=lambda self: str(uuid.uuid4()))
+
+    # to display the warning from specific model
+    share_warning = fields.Text("share warning", compute="_compute_share_warning")
```

Comment:
Just to be coherent shouldn't we name it access_warning so that all portal mixin fields are prefixed by access ?

Reply:
It makes sense ! :+1: